### PR TITLE
docs(cli): add pnpm instructions

### PR DESCRIPTION
**Pull request by soulsam480**
**What kind of change does this PR introduce?** 

- [ ] Bugfix
- [ ] Feature
- [x] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch (or `v[X]` branch)
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [x] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs)  or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to [start a new feature discussion](https://github.com/quasarframework/quasar/discussions/new?category=ideas-proposals) first and wait for approval before working on it)

**Other information:**
As the new Quasar CLI is working properly with pnpm, it makes sense to mention it in the docs. This PR adds pnpm starter commands where npm and yarn are mentioned.

### CODE REVIEW

File: docs/src/pages/quasar-cli-vite/convert-to-quasar-cli-with-vite.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ Quasar CLI with Vite (`@quasar/app-vite`) is currently in **BETA**. Specificatio
 $ yarn create quasar
 # or:
 $ npm init quasar
+# or:
+$ pnpm dx create-quasar
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-vite/convert-to-quasar-cli-with-vite.md]: I guess you meant [`pnpm dlx`](https://pnpm.io/cli/dlx) and not `dx`. But, [`pnpm create`](https://pnpm.io/cli/create) would be simpler and better here:
```suggestion
$ pnpm create quasar
```

File: docs/src/pages/quasar-cli-vite/quasar-config-js.md [hunk 2]
```diff
@@ -174,7 +174,7 @@ preFetch?: boolean;
 
 ### eslint
 
-You will need the linting files already installed. If you don't know which those are, scaffold a new Quasar project folder (`yarn create quasar` or `npm init quasar`) and pick "Linting" when asked about it.
+You will need the linting files already installed. If you don't know which those are, scaffold a new Quasar project folder (`yarn create quasar` or `npm init quasar` or `pnpm dx create-quasar`) and pick "Linting" when asked about it.
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-vite/quasar-config-js.md]:
```suggestion
You will need the linting files already installed. If you don't know which those are, scaffold a new Quasar project folder (`yarn create quasar` or `npm init quasar` or `pnpm create quasar`) and pick "Linting" when asked about it.
```

File: docs/src/pages/quasar-cli-webpack/convert-to-quasar-cli-with-webpack.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ This page will guide you on how to convert a Quasar CLI with Webpack (`@quasar/a
 $ yarn create quasar
 # or:
 $ npm init quasar
+# or:
+$ pnpm dx create-quasar
```
Comment (yusufkandemir) [inline docs/src/pages/quasar-cli-webpack/convert-to-quasar-cli-with-webpack.md]:
```suggestion
$ pnpm create quasar
```

File: docs/src/pages/start/quasar-cli.md [hunk 4]
```diff
@@ -24,6 +24,11 @@ $ yarn create quasar
 
 $ npm i -g @quasar/cli
 $ npm init quasar
+
+# or:
+
+$ pnpm global add @quasar/cli
+$ pnpm dx create-quasar
```
Comment (yusufkandemir) [inline docs/src/pages/start/quasar-cli.md]:
```suggestion
$ pnpm create quasar
```

File: docs/src/pages/start/quasar-cli.md [hunk 5]
```diff
@@ -36,6 +41,8 @@ $ npm init quasar
     $ yarn create quasar
     # or:
     $ npm init quasar
+    # or:
+    $ pnpm dx create-quasar
```
Comment (yusufkandemir) [inline docs/src/pages/start/quasar-cli.md]:
```suggestion
    $ pnpm create quasar
```

File: docs/src/pages/start/quasar-cli.md [hunk 6]
```diff
@@ -24,6 +24,11 @@ $ yarn create quasar
 
 $ npm i -g @quasar/cli
 $ npm init quasar
+
+# or:
+
+$ pnpm global add @quasar/cli
```
Comment (yusufkandemir) [inline docs/src/pages/start/quasar-cli.md]: There is no such thing as `pnpm global`, it should be [`pnpm add -g` or `pnpm add --global`](https://pnpm.io/cli/add#--global--g). It's a mix between `npm`(`-g`) and `yarn`(`add`) 🙂

```suggestion
$ pnpm add -g @quasar/cli
```

File: docs/src/pages/start/quasar-cli.md [hunk 7]
```diff
@@ -53,6 +60,8 @@ $ npm init quasar
     $ yarn global add @quasar/cli
     # or
     $ npm install -g @quasar/cli
+    # or
+    $ pnpm global add @quasar/cli
```
Comment (yusufkandemir) [inline docs/src/pages/start/quasar-cli.md]:
```suggestion
    $ pnpm add -g @quasar/cli
```

**Review by yusufkandemir — CHANGES_REQUESTED** [review]
Thanks for the contribution. It looks like none of the commands are valid, I've made some suggestions.

Aside from that, it might be "working", but it's not properly tested nor officially supported, at least not yet. It shows `Yes, use pnpm` as an answer for `Install project dependencies?`, which might be a bit misleading. But, that's because it's implemented in a generic way on project creation. The big problem is that it would not work when running commands like `quasar mode add`, `quasar new`, `quasar ext add`, etc. which are crucial. The "app core" just assumes you are using either yarn or npm, which are the only officially supported package managers. So, we can't include it on the docs.

File: docs/src/pages/quasar-cli-vite/convert-to-quasar-cli-with-vite.md (hunk 1 above)
Comment (soulsam480) [reply docs/src/pages/quasar-cli-vite/convert-to-quasar-cli-with-vite.md]: Aah yes yes. Will change those. I never tried to run with pnpm create quasar, so it was a mistake from my side. Let me change those.

**Comment by soulsam480** [comment]
I hope the team has some plan to support pnpm considering its speed and efficiency.

File: docs/src/pages/quasar-cli-vite/convert-to-quasar-cli-with-vite.md (hunk 1 above)
Comment (yusufkandemir) [reply docs/src/pages/quasar-cli-vite/convert-to-quasar-cli-with-vite.md]: Fyi, you can easily commit suggestions by using the button.
image

File: docs/src/pages/start/quasar-cli.md [hunk 8]
```diff
@@ -24,6 +24,11 @@ $ yarn create quasar
 
 $ npm i -g @quasar/cli
 $ npm init quasar
+
```
Comment (yusufkandemir) [inline docs/src/pages/start/quasar-cli.md:39]: There is now a tip section called "Requirements" above this. It says "Yarn v1 (strongly recommended) or NPM."

image

Please also update it accordingly.

Comment (yusufkandemir) [reply docs/src/pages/start/quasar-cli.md:39]: I ended up adding this myself into this PR.

**Comment by yusufkandemir** [comment]
Thanks for the contribution again.

**Pull request merged**